src/Persistence_matrix/include/gudhi/matrix.h

```diff
@@ -0,0 +1,664 @@
+{
+	_assert_options();
+}
+
+template<class Options>
+inline Matrix<Options>::Matrix(int numberOfColumns) : matrix_(numberOfColumns)
+{
+	_assert_options();
+}
+
+template<class Options>
+inline Matrix<Options>::Matrix(const Matrix &matrixToCopy) : matrix_(matrixToCopy.matrix_)
+{
+	_assert_options();
+}
+
+template<class Options>
+inline Matrix<Options>::Matrix(Matrix &&other) noexcept : matrix_(std::move(other.matrix_))
+{
+	_assert_options();
+}
+
+template<class Options>
+template<class Boundary_type>
+inline void Matrix<Options>::insert_boundary(Boundary_type &boundary)
+{
+	matrix_.insert_boundary(boundary);
+}
+
+template<class Options>
+inline typename Matrix<Options>::Column_type &Matrix<Options>::get_column(index columnIndex)
+{
+	return matrix_.get_column(columnIndex);
+}
+
+template<class Options>
+inline typename Matrix<Options>::Row_type &Matrix<Options>::get_row(index rowIndex)
+{
+	static_assert(Options::has_row_access, "'get_row' is not implemented for the chosen options.");
+
+	return matrix_.get_row(rowIndex);
+}
+
+template<class Options>
+inline void Matrix<Options>::erase_last()
+{
+	static_assert(Options::has_removable_columns, "'erase_last' is not implemented for the chosen options.");
+
+	matrix_.erase_last();
+}
+
+template<class Options>
+inline dimension_type Matrix<Options>::get_max_dimension() const
+{
+	return matrix_.get_max_dimension();
+}
+
+template<class Options>
+inline unsigned int Matrix<Options>::get_number_of_columns() const
+{
+	return matrix_.get_number_of_columns();
+}
+
+template<class Options>
+inline dimension_type Matrix<Options>::get_column_dimension(index columnIndex) const
+{
+	return matrix_.get_column_dimension(columnIndex);
+}
+
+template<class Options>
+inline void Matrix<Options>::add_to(index sourceColumnIndex, index targetColumnIndex)
+{
+	return matrix_.add_to(sourceColumnIndex, targetColumnIndex);
+}
+
+template<class Options>
+inline void Matrix<Options>::zero_cell(index columnIndex, index rowIndex)
+{
+	return matrix_.zero_cell(columnIndex, rowIndex);
+}
+
+template<class Options>
+inline void Matrix<Options>::zero_column(index columnIndex)
+{
+	return matrix_.zero_column(columnIndex);
+}
+
+template<class Options>
+inline bool Matrix<Options>::is_zero_cell(index columnIndex, index rowIndex)
+{
+	return matrix_.is_zero_cell(columnIndex, rowIndex);
+}
+
+template<class Options>
+inline bool Matrix<Options>::is_zero_column(index columnIndex)
+{
+	return matrix_.is_zero_column(columnIndex);
+}
+
+template<class Options>
+inline index Matrix<Options>::get_column_with_pivot(index simplexIndex)
+{
+	return matrix_.get_column_with_pivot(simplexIndex);
+}
+
+template<class Options>
+inline index Matrix<Options>::get_pivot(index columnIndex)
+{
+	return matrix_.get_pivot(columnIndex);
+}
+
+template<class Options>
+inline Matrix<Options>& Matrix<Options>::operator=(Matrix other)
+{
+	std::swap(matrix_, other.matrix_);
```

Comment:
```suggestion
        using std::swap;
	swap(matrix_, other.matrix_);
```
if you want it to use the specific swap function that you defined for the matrix class.